internal/controller/utils.go

```diff
@@ -16,6 +17,35 @@ func updateDeploymentAnnotations(deployment *appsv1.Deployment, annotations map[
 	}
 }
 
+// syncDeploymentReplicas synchronizes the replicas field between two Deployment objects.
+func syncDeploymentReplicas(a, b *appsv1.Deployment) bool {
+	if a == nil || b == nil {
+		return false
+	}
+
+	if *a.Spec.Replicas != *b.Spec.Replicas {
+		a.Spec.Replicas = b.Spec.Replicas
+		return true
+	}
+
+	return false
+}
+
+func syncDeploymentResources(a, b *appsv1.Deployment) bool {
```

Comment:
`syncDeploymentResources` sounds like it is going to sync every container in the deployment, while here it updates only the first container. Though it is enough for now, if in future we add new containers in different order, the function cannot generalize for every deployment. 
We can use container name as key to merge 2 container lists.

Reply:
We may just need a function to update the resource settings in a container lists of a deployment: 
`func setDeploymentContainerResources(deployment *appsv1.Deployment, resources *corev1.ResourceRequirements, containerName string) error`